### PR TITLE
Add Verification Url to Welcome Email

### PR DESCRIPTION
**Pull request by michael-alade**
**Resolves**
#2125 

**Testing**
1. Clone from dev
1. Add email provider on dashboard
1. Sign as a new user
1. Follow the get started button on the welcome email
1. Observe that the user's document in the database is now showing `true` for the verified field

### CODE REVIEW

**Comment by zenweasel** [comment]
So I seem to be getting two welcome emails now.
inbox__1__-_brent_reactioncommerce_com_-_reaction_commerce_mail

**Comment by aaronjudd** [comment]
@zenweasel I'm not sure that's new. I've been seeing that as well in my recent testing.

**Comment by zenweasel** [comment]
After following the link and setting my password the verified field still shows as `false`.

robomongo_0_9_0-rc9

**Comment by zenweasel** [comment]
I verified that you get the two emails on `development` as well.

Well, this is a bit of a mess. So when I invite a user I get two emails, one that allows me to set my password and one that supposedly verifies my email, except it doesn't actually verify my email. Since the first one allows me to set my password and I can't login until I set my password it seems like that's the one that we want and this second serves no purpose.

File: client/modules/accounts/templates/verify/verifyAccount.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Meteor } from "meteor/meteor";
+import { Template } from "meteor/templating";
+import { Reaction } from "/client/api";
+import { Accounts } from "/lib/collections";
```
Comment (zenweasel) [inline client/modules/accounts/templates/verify/verifyAccount.js]: Unused import

**Comment by zenweasel** [comment]
The issue is with this is that the email in the link is incorrect. The link sent to `brent@reactioncommerce.com` looks like this:
`http://localhost:3000//reaction/account/profile/verify?email=brent@hoover.net`

**Comment by zenweasel** [comment]
The `verified` field in the `user` collection now shows verified, but the field in the `Accounts` collection still shows `false`. Not sure why we have both of these fields or which one is important. @spencern's original ticket does not specify either.

**Comment by spencern** [comment]
@zenweasel I don't think I was aware of the field in Accounts. Probably should set both to `true`

**Comment by aaronjudd** [comment]
Not 100% sure about this, but I think `verified` probably only exists in `Accounts` because when the user is first created, the user record is clone to the Accounts collection.. ? (thus before verification..)

**Comment by zenweasel** [comment]
When I click on the email I am getting a 404 page. I think this was redirecting me to the home page before (could be React Router related?)

Steps to Replicate:

1. `reaction reset`
1. Click on "Sign in" from the main menu
1. Enter email and password
1. Open email and click on "Get Started" link
1. Observe "Oops! This page was not found"

File: server/methods/accounts/accounts.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import _ from "lodash";
 import moment from "moment";
+import { Meteor } from "meteor/meteor";
```
Comment (zenweasel) [inline server/methods/accounts/accounts.js]: Import order

File: server/methods/accounts/accounts.js [hunk 3]
```diff
@@ -145,6 +146,35 @@ function validateAddress(address) {
  * Reaction Account Methods
  */
 Meteor.methods({
```
Comment (zenweasel) [inline server/methods/accounts/accounts.js]: So the style that we want to use for Meteor methods is to `export` them as individual method, and then just declare them as Meteor methods at the bottom of the file. This allows them to more easily be tested or used by other server-side code.

**Comment by zenweasel** [comment]
It looks like this flag is still not being set correctly.

robomongo_0_9_0-rc9

**Comment by aaronjudd** [comment]
@zenweasel @michael-alade what is the status of this PR?

**Comment by michael-alade** [comment]
@zenweasel The changes have been made, and it's ready for another review.

File: server/methods/accounts/accounts.js [hunk 4]
```diff
@@ -8,10 +9,46 @@ import { Logger, Reaction } from "/server/api";
...
+  }
+  return false;
+}
+
+/**
  * @summary Returns the name of the geocoder method to use
  * @returns {string} Name of the Geocoder method to use
  */
-function getValidator() {
+export function getValidator() {
```
Comment (zenweasel) [inline server/methods/accounts/accounts.js]: these functions were purposefully not exported as they are not part of the API

File: server/methods/accounts/accounts.js [hunk 5]
```diff
@@ -141,65 +178,147 @@ function validateAddress(address) {
...
+      Accounts.update({
+        "userId": userId,
+        "profile.addressBook.isBillingDefault": true
+      }, {
+        $set: {
+          "profile.addressBook.$.isBillingDefault": false
         }
-      }
-      // then change the address that has been affected
+      });
```
Comment (zenweasel) [inline server/methods/accounts/accounts.js]: While I support moving these into a more standard format, it is best practice to limit your PR to just changes that affect the issue you are trying to fix (it's in our docs and in the PR template). These changes make it difficult to see what is affecting this issue and what is "refactoring". It also makes it difficult for me to know what to test now.

**Comment by zenweasel** [comment]
This flag is still not being set correctly. This is now the fourth time I have noted this.

robomongo_0_9_0-rc9

**Comment by aaronjudd** [comment]
Is this [WIP]?

**Comment by michael-alade** [comment]
@aaronjudd I'm just making a few changes before I have one of the team members test to make sure it works for them.

File: server/methods/accounts/accounts.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import { Meteor } from "meteor/meteor";
```
Comment (michael-alade) [inline server/methods/accounts/accounts.js]: I've made changes to the import order on this file, but it doesn't seem to be reflecting on this comment.
https://github.com/reactioncommerce/reaction/blob/michael-verify-email-address-2125/server/methods/accounts/accounts.js

**Comment by zenweasel** [comment]
Testing this with registering a user this works, but inviting a user still does not set the `Account` to `true`. The original replication steps were vague about how exactly to create a user but it should be for either method.

**Comment by michael-alade** [comment]
I am currently working to make sure the same verification functionality works when a user is invited to the shop from the dashboard.

**Comment by michael-alade** [comment]
@zenweasel The email verification for an invited user works fine now.

**Pull request merged**